internal/provider/data_host.go

```diff
@@ -0,0 +1,473 @@
+			"frequency":    core.FrequencyMHz,
+			"manufacturer": core.Manufacturer,
+			"model":        core.Model,
+		}
+		cpuCores = append(cpuCores, cpuCore)
+	}
+
+	cpu := map[string]interface{}{
+		"cores":              host.CPU.Cores,
+		"cpu_cores":          cpuCores,
+		"frequency_mhz":      host.CPU.FrequencyMHz,
+		"used_frequency_mhz": host.CPU.UsedFrequencyMHz,
+	}
+
+	_ = d.Set("cpu", cpu)
+
+	// Domain information.
+	domain := []map[string]interface{}{
+		{
+			"id":   host.Domain.ID,
+			"name": host.Domain.Name,
+		},
+	}
+	_ = d.Set("domain", domain)
+
+	// ESXi version information.
+	_ = d.Set("version", host.EsxiVersion)
+
+	// Fully qualified domain name information.
+	_ = d.Set("fqdn", host.Fqdn)
+
+	// Hardware information.
+	hardware := []map[string]interface{}{
+		{
+			"model":  host.HardwareModel,
+			"vendor": host.HardwareVendor,
+			"hybrid": host.Hybrid,
+		},
+	}
+
+	_ = d.Set("hardware", hardware)
+
+	// IP addresses information.
+	ipAddresses := []map[string]interface{}{}
+	for _, ip := range host.IPAddresses {
+		ipAddress := map[string]interface{}{
+			"ip_address": ip.IPAddress,
+			"type":       ip.Type,
+		}
+		ipAddresses = append(ipAddresses, ipAddress)
+	}
+	_ = d.Set("ip_addresses", ipAddresses)
+
+	// Memory information.
+	memory := []map[string]interface{}{
+		{
+			"total_capacity_mb": host.Memory.TotalCapacityMB,
+			"used_capacity_mb":  host.Memory.UsedCapacityMB,
+		},
+	}
+	_ = d.Set("memory", memory)
+
+	// Network pool information.
+	networkPool := []map[string]interface{}{
+		{
+			"id":   host.Networkpool.ID,
+			"name": host.Networkpool.Name,
+		},
+	}
+	_ = d.Set("network_pool", networkPool)
+
+	// Physical NICs information.
+	physicalNics := []map[string]interface{}{}
+	for _, nic := range host.PhysicalNics {
+		physicalNic := map[string]interface{}{
+			"device_name": nic.DeviceName,
+			"mac_address": nic.MacAddress,
+			"speed":       nic.Speed,
+			"unit":        nic.Unit,
+		}
+		physicalNics = append(physicalNics, physicalNic)
+	}
+	_ = d.Set("physical_nics", physicalNics)
+
+	// Status information.
+	_ = d.Set("status", host.Status)
+
+	// Storage information.
+	disks := []map[string]interface{}{}
+	for _, disk := range host.Storage.Disks {
+		diskInfo := map[string]interface{}{
+			"capacity_mb":  disk.CapacityMB,
+			"disk_type":    disk.DiskType,
+			"manufacturer": disk.Manufacturer,
+			"model":        disk.Model,
+		}
+		disks = append(disks, diskInfo)
+	}
+
+	storage := []map[string]interface{}{
+		{
+			"total_capacity_mb": host.Storage.TotalCapacityMB,
+			"used_capacity_mb":  host.Storage.UsedCapacityMB,
+			"disks":             disks,
+		},
+	}
+	_ = d.Set("storage", storage)
+
+	return nil
+}
+
+func getHostByFqdn(ctx context.Context, apiClient *client.VcfClient, fqdn string) (*models.Host, error) {
+	params := hosts.NewGetHostParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+
+	hostResponse, err := apiClient.Hosts.GetHost(params)
```

Comment:
How does the GetHost endpoint work in this setup. Its supposed to accept a host ID and since you don't have that I imagine that this would always fail.
You probably need to switch to GetHosts and iterate over the response

Reply:
Done!